### PR TITLE
avoid code duplication for disabling parallelization with xUnit

### PR DESCRIPTION
**Pull request by adamsitnik**
Few of the tests projects were defining:

```cs
[CollectionDefinition("NoParallelTests", DisableParallelization = true)]
public partial class NoParallelTests { }
```

I've removed the duplications, introduced a single type for it and used `nameof` instead of magic strings to make sure that compiler gives us an error if the type is not referenced in the project.

### CODE REVIEW

**Review by carlossanlop — COMMENTED** [review]
Seems like the new cs file needs to be referenced in additional projects?

```
src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.Http2KeepAlivePing.cs(16,24): 
error CS0103: The name 'NonParallelTestCollection' does not exist in the current context
```

**Comment by adamsitnik** [comment]
> Seems like the new cs file needs to be referenced in additional projects?

**Comment by adamsitnik** [comment]
> Seems like the new cs file needs to be referenced in additional projects?

You are right, I've fixed a build failure and tested it locally (it's not part of `-subset clr+libs+libs.tests` to my suprise).

**Comment by adamsitnik** [comment]
The CI has failed and it's most likely caused by:

> Fixtures can be shared across assemblies, but collection definitions must be in the same assembly as the test that uses them.

Source: https://xunit.net/docs/shared-context#collection-fixture

**Comment by adamsitnik** [comment]
I've fixed the CI failures, PTAL again

File: src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj [hunk 1]
```diff
@@ -152,7 +152,9 @@
     <Compile Include="Design\DesignerTransactionTests.cs" />
     <Compile Include="Security\Authentication\ExtendedProtection\ExtendedProtectionPolicyTypeConverterTests.cs" />
     <Compile Include="XTypeDescriptionProviderTests.cs" />
-
+    <!-- The collection definitions must be in the same assembly as the test that uses them -->
```
Comment (wfurt) [inline src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:155]: I would probably skip the comment duplication. It seems like with this change there will be enough examples to follow and missing reference will result in clear build failure. And there is explanation in the DisableParallelization.cs

**Review by wfurt — APPROVED** [review]
LGTM.

File: src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj [hunk 2]
```diff
@@ -153,6 +153,8 @@
     <Compile Include="Security\Authentication\ExtendedProtection\ExtendedProtectionPolicyTypeConverterTests.cs" />
     <Compile Include="XTypeDescriptionProviderTests.cs" />
 
+    <Compile Include="$(CommonTestPath)TestUtilities\System\DisableParallelization.cs"
+        Link="Common\TestUtilities\System\DisableParallelization.cs" />
```
Comment (safern) [inline src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:157]: Why not just include it as part of `TestUtilities.csproj` which is referenced by default on all test projects?

Comment (adamsitnik) [reply src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:157]: > Why not just include it as part of TestUtilities.csproj which is referenced by default on all test projects?

@safern this is what I've tried in the beginning and got a lot of new CI failures.

From https://xunit.net/docs/shared-context#collection-fixture I've learned that:

> Fixtures can be shared across assemblies, but collection definitions must be in the same assembly as the test that uses them.

So we have to reference the file in every project that wants to use it ;(

Comment (safern) [reply src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:157]: Ah i see, that's unfortunate.

**Pull request merged**